Formula/youtube-viewer.rb

```diff
@@ -0,0 +1,147 @@
+
+  resource "LWP::Protocol::https" do
+    url "https://cpan.metacpan.org/authors/id/O/OA/OALDERS/LWP-Protocol-https-6.07.tar.gz"
+    sha256 "522cc946cf84a1776304a5737a54b8822ec9e79b264d0ba0722a70473dbfb9e7"
+  end
+
+  resource "LWP::UserAgent" do
+    url "https://cpan.metacpan.org/authors/id/O/OA/OALDERS/libwww-perl-6.39.tar.gz"
+    sha256 "9a8d7747938aa75d7d524c71574ae7f99ca66a5dac8255a7f7759f373e7774d5"
+  end
+
+  resource "List::Util" do
+    url "https://cpan.metacpan.org/authors/id/P/PE/PEVANS/Scalar-List-Utils-1.52.tar.gz"
+    sha256 "279d78cef84acae280da4dfb95eff0c9865d1611b1a3b026baddf42d1ba01de4"
+  end
+
+  resource "MIME::Base64" do
+    url "https://cpan.metacpan.org/authors/id/G/GA/GAAS/MIME-Base64-3.15.tar.gz"
+    sha256 "7f863566a6a9cb93eda93beadb77d9aa04b9304d769cea3bb921b9a91b3a1eb9"
+  end
+
+  resource "Term::ANSIColor" do
+    url "https://cpan.metacpan.org/authors/id/R/RR/RRA/Term-ANSIColor-4.06.tar.gz"
+    sha256 "8161c7434b1984bde588d75f22c786c46cb6d35d264d58111db0b82537de4bad"
+  end
+
+  resource "Term::ReadLine" do
+    url "https://cpan.metacpan.org/authors/id/F/FL/FLORA/Term-ReadLine-1.14.tar.gz"
+    sha256 "54523c72b26a04608170849013a433ba400f66b4f9b0524201bfd37ff6e3c477"
+  end
+
+  resource "Text::ParseWords" do
+    url "https://cpan.metacpan.org/authors/id/C/CH/CHORNY/Text-ParseWords-3.30.tar.gz"
+    sha256 "85e0238179dd43997e58c66bd51611182bc7d533505029a2db0d3232edaff5e8"
+  end
+
+  resource "Text::Wrap" do
+    url "https://cpan.metacpan.org/authors/id/M/MU/MUIR/modules/Text-Tabs+Wrap-2013.0523.tar.gz"
+    sha256 "b9cb056fffb737b9c12862099b952bf4ab4b1f599fd34935356ae57dab6f655f"
+  end
+
+  resource "URI::Escape" do
+    url "https://cpan.metacpan.org/authors/id/O/OA/OALDERS/URI-1.76.tar.gz"
+    sha256 "b2c98e1d50d6f572483ee538a6f4ccc8d9185f91f0073fd8af7390898254413e"
+  end
+
+  # optional runtime deps
+  resource "LWP::UserAgent::Cached" do
+    url "https://cpan.metacpan.org/authors/id/O/OL/OLEG/LWP-UserAgent-Cached-0.07.tar.gz"
+    sha256 "4a9cdc47e58560e39dd99a08d1dbddcbc998942ebcdc9deca6e2b220b0d86ae0"
+  end
+
+  resource "Mozilla::CA" do
+    url "https://cpan.metacpan.org/authors/id/A/AB/ABH/Mozilla-CA-20180117.tar.gz"
+    sha256 "f2cc9fbe119f756313f321e0d9f1fac0859f8f154ac9d75b1a264c1afdf4e406"
+  end
+
+  def install
+    ENV.append "PERL_MM_USE_DEFAULT", 1
+    ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+
+    resources.each do |r|
+      r.stage do
+        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+        system "make"
+        system "make", "install"
+      end
+    end
+
+    ENV.prepend_path "PATH", libexec/"bin"
+
+    system "perl", "Build.PL", "--install_base", libexec
+    system "./Build", "install"
+
+    %w[youtube-viewer].each do |cmd|
+      (bin/cmd).write_env_script(libexec/"bin/#{cmd}", :PERL5LIB => ENV["PERL5LIB"])
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/youtube-viewer --version").split(/ /)[2]
```

Comment:
Is there any way to test the functionality of this software rather than just printing the version number ?

Reply:
Is this tool only interactive?

Reply:
@javian @SMillerDev is this good as-is or do I need to make changes here to get this accepted?

Reply:
There needs to be some test. Maybe you could do it like the apache tests where you run and then kill it and check some output?

Reply:
Is... that not what I'm already doing? Run `youtube-viewer --version`, let it produce output then quit, capture the output and ensure it has the correct value.

Reply:
please test somebody on Ubuntu >= 19.04 I have Ubuntu 19.04 and got:
```
make
Last 15 lines from /home/salessionato/.cache/Homebrew/Logs/youtube-viewer/17.make:
Reading MacThai (MacThai)
Reading MacTurkish (MacTurkish)
Reading MacUkrainian (MacUkrainian)
Reading nextstep (nextstep)
Reading viscii (viscii)
"/usr/bin/perl" "/usr/share/perl/5.28.1/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/share/perl/5.28/ExtUtils/typemap'  Byte.xs > Byte.xsc
x86_64-linux-gnu-gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=declaration-after-statement -Wpointer-sign -O2 -g   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.28/CORE"   byte_t.c
mv Byte.xsc Byte.c
x86_64-linux-gnu-gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=declaration-after-statement -Wpointer-sign -O2 -g   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.28/CORE"   Byte.c
Byte.c:10:20: fatal error: EXTERN.h: No such file or directory
compilation terminated.
make[1]: *** [Makefile:322: Byte.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/tmp/youtube-viewer--Encode-20191125-5568-1ipxfa9/Encode-3.01/Byte'
make: *** [Makefile:649: subdirs] Error 2
```

- which -a perl shows first lonuxbrew's perl

- ~/.cpan removed

- local:lib in $HOME/perl5 removed and recreated with source ~/.bashrc

Reply:
@JuPlutonic this repository doesn't support Linux. file an issue on linuxbrew-core.

Reply:
@MikeMcQuaid sorry, could you clarify what action you want here? Do I need to create a companion PR on `linuxbrew-core` adding this formula there, as well, before we can get this one merged here?

Reply:
He was responding to a comment by someone else that has since been deleted.

Reply:
Ah, my bad -- thanks @Bo98 !

Reply:
No, youtube-viewer will misbehave if something happen to perl/SSL-certificates or to Youtube's servers (and their GET queries and hypermedia, if presented, seems to me are unchangeable).

Do the perl testing is not needed at all in this formula and do smth like opening the app interactively and producing queries is nonsense -
 we haven't have instruments like `brew test formula --listen_port=8080 --listen_url=0.0.0.0 --query_type=get`

> There needs to be some test. Maybe you could do it like the apache tests where you run and then kill it and check some output?

Reply:
> youtube-viewer will misbehave if something happen to perl/SSL-certificates(I mean the y-v dependencies) or to Youtube's servers (and their GET queries and hypermedia, if presented, seems to me are unchangeable).

This is desirable: we want the test to fail in this case.

Reply:
> youtube-viewer [tests] will misbehave if something happen to perl/SSL-certificates(I mean the y-v dependencies)

That seems desireable...

> or to Youtube's servers (and their GET queries and hypermedia, if presented, seems to me are unchangeable)

...but that does not; that'll be very flaky tests, most likely. Do we need to build a mock YouTube server for this?

Reply:
> Do we need to build a mock YouTube server for this?

No. 

> that'll be very flaky tests, most likely

I disagree. YouTube's uptime is extremely high. We would rather tests that are periodically flaky (and can be rerun) to those that do not exercise the actual functionality of the application.